providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/Operator.java

```diff
@@ -53,34 +59,37 @@ public Object apply(final String flagKey, final String targetingRule, final Eval
     @Getter
     static class FlagProperties {
         private final String flagKey;
+        private final String timestamp;
         private final String targetingKey;
 
         FlagProperties(Object from) {
             if (from instanceof Map) {
                 Map<?, ?> dataMap = (Map<?, ?>) from;
 
-                final Object flagKey = Optional.ofNullable(dataMap.get(FLAGD_PROPS_KEY))
-                    .filter(flagdProps -> flagdProps instanceof Map)
-                    .map(flagdProps -> ((Map<?, ?>)flagdProps).get(FLAG_KEY))
-                    .orElse(null);
-
-                if (flagKey instanceof String) {
-                    this.flagKey = (String) flagKey;
-                } else {
-                    this.flagKey = null;
-                }
+                this.flagKey = extractSubPropertyFromFlagd(dataMap, FLAG_KEY);
+                this.timestamp = extractSubPropertyFromFlagd(dataMap, TIME_STAMP);
 
                 final Object targetKey = dataMap.get(TARGET_KEY);
-
                 if (targetKey instanceof String) {
                     targetingKey = (String) targetKey;
                 } else {
                     targetingKey = null;
                 }
+                
             } else {
                 flagKey = null;
+                timestamp = null;
                 targetingKey = null;
             }
         }
+
+        private static String extractSubPropertyFromFlagd(Map<?, ?> dataMap, String propertyName) {
+            return Optional.ofNullable(dataMap.get(FLAGD_PROPS_KEY))
+                    .filter(flagdProps -> flagdProps instanceof Map)
+                    .map(flagdProps -> ((Map<?, ?>) flagdProps).get(propertyName))
+                    .filter(value -> value instanceof String)
+                    .map(String::valueOf)
+                    .orElse(null);
```

Comment:
This always return null for timestamp as we are storing it as long. Needs to fix this and also I would like to see a simple fractional test (as this is the only place right now that we use these context values) . See a sample test here - https://github.com/open-feature/java-sdk-contrib/blob/main/providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/OperatorTest.java#L37

Reply:
We are directly using the timestamp [here](https://github.com/open-feature/java-sdk-contrib/pull/512/files#diff-3c2fc3d40e2a528288cb6e14d60ead19aa8f2ad3ff2d328b0e04fa09cf14a1d9R38-R58) in this new test.

I think you're right that we're missing a test case, but I think it would have to be a test case that manually runs `extractSubPropertyFromFlagd`, not one that uses some kind of fractional rule (I don't know how you'd use the timestamp IN a fractional rule; only the flag key is used by that rule implementation, which we did not change).

Basically, to avoid confusion, I suggested @DBlanchard88  extend the `extractSubPropertyFromFlagd` method to include this new prop, but that extracted prop is not used anywhere by other rules - we probably should just directly test it (which we weren't before).

Reply:
@toddbaert agrees with the suggestion.

Reply:
> I don't know how you'd use the timestamp IN a fractional rule; only the flag key is used by that rule implementation, which we did not change

@toddbaert on a second thought, with this background, should we even attempt to expose timestamp through `FlagProperties` ? The confusion for me came from the usage of timestamp at `FlagProperties`. But like you said we don't even need it there 🤔  Json logic can expand the value when this property get expanded in other references.

Reply:
It's just that the name of the method seems to indicate that it will extract all the properties. I would use it assuming that, which might cause bugs or confusion if I did... maybe at some point we implement some "time-based rollout" rule - I might want to use the timestamp in that rule and then be confused why that prop is missing.

Reply:
that's a fair point. Then let's stick to your proposal - improve this method to extract the newly added timestamp and add a simple validation test